### DISCUSSION TITLE
How can I use pyaudio for Streaming instead of aplay

### DISCUSSION THREAD

**Discussion by JahanHH** [rhasspy/piper#336] [Q&A]
Unfortunately i have the problem that i can't use 'aplay' under windows because it can only be installed via linux (maybe i'm wrong).
But I can use pyaudio under windows.
Is it possible to use pyaudio instead of aplay for streaming?
If so, can someone give me an example of what really works?

**Comment by jmtatsch** [comment]
https://superuser.com/questions/1024792/is-there-a-windows-equivalent-to-the-linux-aplay-that-will-accept-a-bitstream

**Comment by Cdddo** [comment]
I'm sure there's a way to use pyaudio but I haven't got that one down yet, but I did find out how to do it in Windows with VLC or ffplay (usually comes with ffmpeg). It took me FAR too long to figure this out, but I eventually got it. Make sure you put Piper and VLC (or ffplay) in your PATH and then you can do the following commands in a terminal.

This will synthesize the audio and stream the playback with VLC without making a file:
```
echo "This is piper." | piper -m "C:\your\voice\path\voice.onnx" --output-raw | vlc -I dummy --demux=rawaud --rawaud-samplerate 22050 --rawaud-channels=1 --rawaud-fourcc=s16l - vlc://quit
```

This will generate a file and then play it with VLC:
```
echo "This is piper." | piper -m "C:\your\voice\path\voice.onnx" -f "C:\your\output\path\filename.wav"
vlc -I dummy "C:\your\output\path\filename.wav" vlc://quit
```

Alternatively you can use ffplay to stream:
```
echo "This is piper." | piper -m "C:\your\voice\path\voice.onnx" --output-raw | ffplay -f s16le -ar 22050 -ac 1 -nodisp -autoexit -
```
